src/ServiceProvider.php

```diff
@@ -73,16 +73,56 @@ public function boot()
         });
     }
 
+    protected function legacyPermissionKey($permission, $resource)
+    {
+        return match ($permission) {
+            'view' => "View {$resource->plural()}",
+            'edit' => "Edit {$resource->plural()}",
+            'create' => "Create new {$resource->singular()}",
+            'delete' => "Delete {$resource->singular()}"
+        };
+    }
+
+    protected function permissionKey($permission, $resource)
```

Comment:
Out of interest, why do these permission keys need to be configurable, can't they just be fixed? Isn't it just the labels that need to be translated?

For example: it's just `view article`/`create article`, where `article` is the handle of the resource?

Reply:
In my example, I'm referring to the names of permissions, not the labels (which is what's seen in the Control Panel).

Reply:
Yes... I understood that. And I also explained why the **names** must also be configurable, not only the **labels**.

`$user->hasPermission('View Autores')` or `$user->hasPermission('Create new Autore')` are examples of code that I have to write currently. `View Autores` and `Create new Autore` are the **NAMES** currently being generated by Runway, **NOT LABELS**.

**CURRENTLY:**
Name: 'Create new Autore'
Label: 'Create new Autore'

**WITH THIS PR WITHOUT CONFIGURING ANYTHING:**
Name: 'Create new Autore'
Label: 'Create new Autore'

**WITH THIS PR CONFIGURING **ONLY THE LABELS**:**
Name: 'Create new Autore'
Label: 'Criar autor'

**WITH THIS PR CONFIGURING **NAMES AND LABELS**:**
Name: 'create author'
Label: 'Criar autor'

Reply:
Thanks, I think I understand now.

Instead of making those names configurable (like you have them now in the `runway.php` config file), can you hard code these names into Runway instead?

Statamic doesn't make them configurable for its permissions.

![image](https://user-images.githubusercontent.com/19637309/216361030-5a32ba60-0a47-4c6d-8f9b-9d539687c1c6.png)

Reply:
Well... We could do that... But my thinking was that people already using the package had their permission names already stored in their stache files or their databases. Because of this, hardcoding this new permission names will cause their permissions to break when updating the package, because they will have permissions like "Create new Autore" saved on their stores, but the package will check for a "create author" permission.

That's why I chose the configurable path. Hardcoding this will have to come on a major release through a breaking change and, even then, maybe keeping this configurable would better.

Reply:
Don't worry about that - I can add a migration script to change the permissions when folks update.

Reply:
IMO this is quite risky... But if you think this won't blow things up... Sure... We can try that...

Reply:
It should be fine. It'll be included in a breaking release & will have a note in the changelog if people need to do it manually.